src/renderer/services/dex.js

```diff
@@ -1679,15 +1680,29 @@ export default {
               params: [assets.DEX_SCRIPT_HASH, `${u.reverseHex(assets.APH)}fc`],
             })
               .then((res) => {
-                if (!res || !res.result || res.result.length < 32) {
-                  dexState.totalFeesCollected = 0;
-                  resolve(dexState);
-                  return;
+                dexState.totalFeesCollected = 0;
+
+                if (res.result && res.result.length >= 32) {
+                  dexState.totalFeesCollected = u.fixed82num(res.result.substr(0, 16));
                 }
 
-                dexState.totalFeesCollected = u.fixed82num(res.result.substr(0, 16));
-                // console.log(`dexState.totalFeesCollected: ${dexState.totalFeesCollected}`);
-                resolve(dexState);
+
+                rpcClient.query({
+                  method: 'getstorage',
+                  params: [assets.DEX_SCRIPT_HASH, u.str2hexstring('claimMinimumBlocks')],
+                })
+                  .then((res) => {
+                    dexState.minimumClaimBlocks = claiming.DEFAULT_CLAIM_BLOCKS;
+
+                    if (res.result) {
+                      dexState.minimumClaimBlocks = u.fixed82num(res.result);
+                    }
+
+                    resolve(dexState);
+                  })
+                  .catch((e) => {
```

Comment:
We should probably tailed the first sentence for all these errors more specifically to what is failing. Like for this one it is a failure to fetch the minimum claim blocks.

Also, if it fails to get the minimum claim blocks from storage, should it just fall back to the default claim blocks and resolve the dex state rather than reject?

Reply:
Updated